CMakeLists.txt

```diff
@@ -2,93 +2,173 @@ cmake_minimum_required (VERSION 3.5)
 
 project (korc_proj C CXX Fortran)
 
-#  Set directories to build all binary files.
-set (EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/build/bin)
-
-#  Fortran specific settings.
-set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -cpp")
-set (CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/modules)
-set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/lib)	
+#  FIXME: Remove once link issues are resolved with the new macos linker.
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+   SET (CMAKE_EXE_LINKER_FLAGS -ld_classic)
+endif ()
 
-#  Set definitions.
-add_definitions (-DDOUBLE_PRECISION)
-add_definitions (-DPARALLEL_RANDOM)
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-#   add_definitions (-DDBG_CHECK)
-endif()
+#-------------------------------------------------------------------------------
+#  Setup build types.
+#-------------------------------------------------------------------------------
+set_property (CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+              Debug
+              Release
+)
```

Comment:
```suggestion
```

Reply:
Why would we want to allow non Release or Debug build types?

Reply:
One goal is for KORC to build like any other CMake project. Modifying default build types without a strong reason for doing so just creates more challenges when working on multiple projects.

Personally, I often use `RelWithDebInfo` so I can still get a backtrace and step through with gdb.
 
One possible improvement in issue #15